diff -uprN dnsdist-1.5.0-rc2.old/ext/lmdb-safe/lmdb-safe.hh dnsdist-1.5.0-rc2/ext/lmdb-safe/lmdb-safe.hh
--- dnsdist-1.5.0-rc2.old/ext/lmdb-safe/lmdb-safe.hh	2020-03-23 15:20:46.000000000 +0300
+++ dnsdist-1.5.0-rc2/ext/lmdb-safe/lmdb-safe.hh	2020-06-16 18:01:32.620843835 +0300
@@ -12,8 +12,9 @@
 #include <vector>
 #include <algorithm>
 
-// apple compiler somehow has string_view even in c++11!
-#if __cplusplus < 201703L && !defined(__APPLE__) && !defined(__OpenBSD__) && !defined(__FreeBSD__)
+#ifdef __cpp_lib_string_view
+using std::string_view;
+#else
 #include <boost/version.hpp>
 #if BOOST_VERSION >= 106100
 #include <boost/utility/string_view.hpp>
@@ -22,8 +23,6 @@ using boost::string_view;
 #include <boost/utility/string_ref.hpp>
 using string_view = boost::string_ref;
 #endif
-#else // C++17
-using std::string_view;
 #endif
 
 
